flink-end-to-end-tests/test-scripts/kafka-common.sh

```diff
@@ -143,7 +146,7 @@ function start_confluent_schema_registry {
 
   if ! get_and_verify_schema_subjects_exist; then
       echo "Could not start confluent schema registry"
-      exit 1
+      return 1
```

Comment:
We are not checking the returned code. Previously, the test would prematurely stop, I think.

Reply:
Yes previously it would exit immediately but that makes it impossible to introduce any retry behavior. (well I suppose a sub-shell might work)
The exit code of test_confluent_schema_registry#test_setup is implicitly the return value of #start_confluent_schema_registry, and hence is evaluated by the retry loop.

Reply:
Right